Content.Server/NodeContainer/Nodes/NodeHelpers.cs

```diff
@@ -9,9 +8,11 @@ namespace Content.Server.NodeContainer.Nodes
     /// </summary>
     public static class NodeHelpers
     {
-        public static IEnumerable<Node> GetNodesInTile(EntityQuery<NodeContainerComponent> nodeQuery, MapGridComponent grid, Vector2i coords)
+        public static IEnumerable<Node> GetNodesInTile(EntityQuery<NodeContainerComponent> nodeQuery, TransformComponent xform, MapGridComponent grid, Vector2i coords)
         {
-            foreach (var entityUid in grid.GetAnchoredEntities(coords))
+            var mapSys = IoCManager.Resolve<IEntitySystemManager>().GetEntitySystem<SharedMapSystem>();
```

Comment:
I understand that you can't do a normal dependency injection on a static class, but resolving the entity system each time the function gets called is likely to be problematic. You should get Sloth's input on this